internal/state.go

```diff
@@ -103,3 +105,49 @@ func (s *StateModel) Watch(cb func(model *StateModel, err error), ctx context.Co
 		cb(s, nil)
 	}
 }
+
+func (s *StateModel) Stop() {
+	s.Watching = false
+}
+
+func (s *StateModel) UpdateMetrics(
+	ctx context.Context,
+	client *api.ClientWithResponses,
+	timings []time.Duration,
+	txns []float64,
+) {
+	if s == nil {
+		panic("StateModel is nil while UpdateMetrics is called")
+	}
+	// Set Metrics
+	s.Metrics.RoundTime = getAverageDuration(timings)
+	s.Metrics.Window = len(timings)
+	s.Metrics.TPS = getAverage(txns)
+
+	// Fetch RX/TX
+	res, err := GetMetrics(ctx, client)
+	if err != nil {
+		s.Metrics.Enabled = false
+	}
+	if err == nil {
+		s.Metrics.Enabled = true
+		s.Metrics.TX = res["algod_network_sent_bytes_total"]
+		s.Metrics.RX = res["algod_network_received_bytes_total"]
+	}
+}
+
+func (s *StateModel) UpdateAccounts() {
+	s.Accounts = AccountsFromState(s)
+}
+
+func (s *StateModel) UpdateKeys(ctx context.Context, client *api.ClientWithResponses) {
+	var err error
+	s.ParticipationKeys, err = GetPartKeys(ctx, client)
+	if err != nil {
+		s.Admin = false
```

Comment:
Is this for the case when we are trying to connect to someone else's node and we're not admins?

Reply:
Connecting to Urtho's mainnet works now